lib/blocktree/leaves.go

```diff
@@ -13,6 +13,7 @@ import (
 
 // leafMap provides quick lookup for existing leaves
 type leafMap struct {
+	sync.RWMutex
```

Comment:
why do we need this mutex if it's a `sync.Map`?

Reply:
there's no way to atomically delete one key from a map and then add another key, eg. in `Replace`:
```
// Replace deletes the old node from the map and inserts the new one
func (ls *leafMap) replace(oldNode, newNode *node) {
	ls.Lock()
	defer ls.Unlock()
	ls.smap.Delete(oldNode.hash)
	ls.store(newNode.hash, newNode)
}
```

it needs to atomically do the replace, otherwise when other functions read the leaf map there might not see either the old node or the new node